src/librustc/session/mod.rs

```diff
@@ -600,11 +600,51 @@ impl Session {
             .panic
             .unwrap_or(self.target.target.options.panic_strategy)
     }
-    pub fn linker_flavor(&self) -> LinkerFlavor {
+
+    fn linker(&self) -> &str {
```

Comment:
AFAIK there's not actually any reason these methods have to live in librustc, if you'd like feel free to delete them here and move the wholesale into librustc_trans

Reply:
I also think this method looks mostly correct except for the last part. We'll probably want to rename this to something like `linker_file_stem` and then make sure you use the `file_stem` method of `Path` at the end to remove bat/exe suffixes